plugins/by-name/openscad/default.nix

```
@@ -1,81 +1,170 @@
+      {
+        old = "cheatsheetToggle";
+        new = "cheatsheet_toggle_key";
+      }
+      {
+        old = "helpTrigger";
+        new = "help_trig_key";
+      }
+      {
+        old = "helpManualTrigger";
+        new = "help_manual_trig_key";
+      }
+      {
+        old = "execOpenSCADTrigger";
+        new = "exec_openscad_trig_key";
+      }
+      "topToggle"
+    ]);
 
-    fuzzyFinder = helpers.defaultNullOpts.mkEnum [
-      "skim"
-      "fzf"
-    ] defaultFuzzyFinder "fuzzy finder to find documentation";
+  settingsOptions = {
+    fuzzy_finder = defaultNullOpts.mkStr "skim" ''
+      Fuzzy finder to find documentation.
 
-    cheatsheetWindowBlend = helpers.defaultNullOpts.mkNullable (types.ints.between 0 100) 15 "";
+      If you set this option explicitly, Nixvim will install the relevant finder plugin.
+    '';
 
-    loadSnippets = helpers.defaultNullOpts.mkBool false "";
+    cheatsheet_window_blend = defaultNullOpts.mkNullable (types.ints.between 0 100) 15 ''
+      Transparency level of the cheatsheet window (in %).
+    '';
 
-    autoOpen = helpers.defaultNullOpts.mkBool false "";
+    load_snippets = defaultNullOpts.mkBool false ''
+      Whether to load predefined snippets for OpenSCAD.
+    '';
 
-    keymaps = {
-      enable = mkEnableOption "keymaps for openscad";
+    auto_open = defaultNullOpts.mkBool false ''
+      Whether the openscad project automatically be opened on startup.
+    '';
 
-      cheatsheetToggle = helpers.defaultNullOpts.mkStr "<Enter>" "Toggle cheatsheet window";
+    default_mappings = defaultNullOpts.mkBool true ''
+      Whether to enable the default mappings.
+    '';
 
-      helpTrigger = helpers.defaultNullOpts.mkStr "<A-h>" "Fuzzy find help resource";
+    cheatsheet_toggle_key = defaultNullOpts.mkStr "<Enter>" ''
+      Keyboard shortcut for toggling the cheatsheet.
+    '';
 
-      helpManualTrigger = helpers.defaultNullOpts.mkStr "<A-m>" "Open offline openscad manual in pdf via zathura";
+    help_trig_key = defaultNullOpts.mkStr "<A-h>" ''
+      Keyboard shortcut for triggering the fuzzy-find help resource.
+    '';
 
-      execOpenSCADTrigger = helpers.defaultNullOpts.mkStr "<A-o>" "Open file in OpenSCAD";
+    help_manual_trig_key = defaultNullOpts.mkStr "<A-m>" ''
+      Keyboard shortcut for manually triggering the offline OpenSCAD manual.
+    '';
 
-      topToggle = helpers.defaultNullOpts.mkStr "<A-c>" "toggle htop filtered for openscad processes";
-    };
+    exec_openscad_trig_key = defaultNullOpts.mkStr "<A-o>" ''
+      Keyboard shortcut for opening the current file in OpenSCAD.
+    '';
+
+    top_toggle = defaultNullOpts.mkStr "<A-c>" ''
+      Keyboard shortcut for toggling `htop` filtered for OpenSCAD processes.
+    '';
+  };
+  settingsExample = {
+    load_snippets = true;
+    fuzzy_finder = "fzf";
+    cheatsheet_window_blend = 15;
+    auto_open = true;
   };
+  settingsDescription = settingsOptionDescription { inherit name globalPrefix; };
```

```suggestion
  };

  settingsDescription = settingsOptionDescription { inherit name globalPrefix; };
```

Also: consider having `settingsDescription` declared at the top of the file, with other trivial metadata?

---

Hmm, not sure about this one. It is solely used there, so I don't see the point of "spreading" the information.
If you strongly prefer moving it, I will do it though.